src/screens/nano/NanoContractDetail.js

```diff
@@ -0,0 +1,203 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, { useContext, useEffect, useState } from 'react';
+import { t } from 'ttag'
+import $ from 'jquery';
+import BackButton from '../../components/BackButton';
+import ReactLoading from 'react-loading';
+import colors from '../../index.module.scss';
+import ModalChangeAddress from '../../components/nano/ModalChangeAddress';
+import helpers from '../../utils/helpers';
+import hathorLib from '@hathor/wallet-lib';
+import { useDispatch, useSelector } from 'react-redux';
+import { useNavigate, useParams } from 'react-router-dom';
+import { addBlueprintInformation } from '../../actions';
+import { get } from 'lodash';
+import { GlobalModalContext, MODAL_TYPES } from '../../components/GlobalModal';
+
+
+/**
+ * Details of a Nano Contract
+ *
+ * @memberof Screens
+ */
+function NanoContractDetail(props) {
+  const context = useContext(GlobalModalContext);
+
+  const { nanoContracts, blueprintsData, tokenMetadata } = useSelector((state) => {
+    return {
+      nanoContracts: state.nanoContracts,
+      blueprintsData: state.blueprintsData,
+      tokenMetadata: state.tokenMetadata
+    }
+  });
+
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+  const { nc_id: ncId } = useParams();
+  const nc = nanoContracts[ncId];
+  let blueprintInformationAux = blueprintsData[nc.blueprintId];
```

Comment:
suggestion: Variables on the higher scope of function components can be misleading about their actual value. If you want predictable values I recommend using states.

For this specific case I'd suggest removing this higher-scope variable and only initializing it right before when it's used, within stricter scopes:
- On the initialization `useEffect()`
- Inside `loadBlueprintInformation()`

Reply:
The reason I need this to be in a variable is because I might get this information from the API response and setting it as a state is async, so I couldn't use in the following API call, then I can't have it inside `loadBlueprintInformation` only. Regarding the `useEffect` scope, it could be possible but the code would be much worse, and I don't see any benefits in doing it, only downsides.

Reply:
Since this is a read-only screen, I believe we can keep this local variable without any greater worries.

However, it's not always intuitive to understand when a functional component will re-render. And, when it does, what consequence this will have on non-`state` and non-`ref` local variables.

As a reference, we've recently had trouble with local variables that were being mutated in a screen on PR #546 and had to choose another approach to fetching and using that local object.